plugins-base/XSFeatureInterface.py

```diff
@@ -175,8 +177,10 @@ def UpdateFieldsPRECOMMIT(self):
                 pane.AddStatusField(Lang("Netmask",  16),  self.netmask)
                 pane.AddStatusField(Lang("Gateway",  16),  self.gateway)
 
-            if self.mode != 'Static' and self.hostname == '':
+            if self.mode == 'DHCP' and self.hostname == '':
                 pane.AddStatusField(Lang("Hostname",  16), Lang("Assigned by DHCP"))
+            elif self.mode == 'Autoconf' and self.hostname == '':
+                pane.AddStatusField(Lang("Hostname", 16), Lang("Automatically assigned"))
```

Comment:
I think both case could be in the same If with a generic sentence about auto assignation.

Reply:
I think "Assigned by DHCP" has a specific meaning. Which does not apply to Autoconf. Making it not appear in Autoconf might be a better solution?